docs/spikes/using-image-data/ai-vision.ipynb

```diff
@@ -0,0 +1,530 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "bbf13a2f",
+   "metadata": {},
+   "source": [
+    "<!-- pip install azure-ai-documentintelligence -->"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6db70b5c",
+   "metadata": {},
+   "source": [
+    "# Using Image Data\n",
+    "\n",
+    "This workbook acts as a spike for how to ingest images and ask questions about the data within them using Azure AI \n",
+    "Vision and GPT-4-Vision.\n",
+    "\n",
+    "## How to use\n",
+    "\n",
+    "To use this you need the following resources provisioned:\n",
+    "- Azure AI Search\n",
+    "- Azure Storage Account\n",
+    "- Azure OpenAI\n",
+    "    - A `text-embeddings-ada-002` model deployment\n",
+    "    - A `gpt-4-vision` model deployment\n",
+    "- Azure Computer Vision\n",
+    "\n",
+    "You can reuse some of the components deployed as part of the Chat With Your Data project. However, at the time of \n",
+    "writing you will also additionally need to deploy Azure Computer Vision and a `gpt-4-vision ` model.\n",
+    "\n",
+    "Once provisioned, you need then to change the configuration below to match your setup.\n",
+    "\n",
+    "## Data\n",
+    "\n",
+    "This spike only uploads one image, which is a flow chart detailing when to use various Azure services. This image was \n",
+    "sourced from https://learn.microsoft.com/en-us/azure/architecture/guide/technology-choices/compute-decision-tree.\n",
+    "\n",
+    "Feel free to use a different image and question.\n",
+    "\n",
+    "## How it works\n",
+    "\n",
+    "### Ingestion\n",
+    "\n",
+    "To ingest an image into Azure AI Search, the following steps are performed:\n",
+    "1. Create search index\n",
+    "2. Upload image to blob storage\n",
+    "3. Generate image embeddings using computer vision\n",
+    "4. Generate a caption of the image using `gpt-4-vision`\n",
+    "5. Generate embeddings of the caption using `text-embeddings-002-ada`\n",
+    "6. Store data in search index\n",
+    "\n",
+    "\n",
+    "### Question\n",
+    "\n",
+    "To ask a question using this data, the following steps are performed:\n",
+    "1. Generate embeddings for the question using `text-embeddings-002-ada`\n",
+    "2. Generate image imbeddings for the question using computer vision\n",
```

Comment:
This second point seems to suggest that the question can be in the form of an image too, not just the data - is this correct? If yes, I'd suggest making this clear in the introduction sentence.

Reply:
Ok, I think we need a little more explanation here. Why do we need to generate embeddings both using the ada model and computer vision?

Reply:
What I would've thought is, since we're finally creating embeddings for the ingested image using the ada model (after the internediate steps of captioning etc), it should suffice to create the ada embeddings for the text question in order to make searching possible. Let me know what I'm missing here.

Reply:
Ah so the `vectorizeImage` computer vision API uses a different embedding model then `text-embeddings-ada-002`. So to search for that image using a text based question, you need to use the `vectorizeText` computer vision API. You can't use `text-embeddings-ada-002` to embed the question and search for images as the vectors are constructed differently, including the size.

Using `gpt-4-vision` to generate a description and `text-embeddings-ada-002` to embed that description is not required, but provides extra context and better results, which I found especially useful for the flow chart.

I'll add a section explaining this now.